apps/src/lib/levelbuilder/SaveBar.jsx

```diff
@@ -1,71 +1,82 @@
 import PropTypes from 'prop-types';
-import React, {Component} from 'react';
+import React from 'react';
 import color from '@cdo/apps/util/color';
 import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import {navigateToHref} from '@cdo/apps/utils';
 
-export default class SaveBar extends Component {
-  static propTypes = {
-    lastSaved: PropTypes.number,
-    error: PropTypes.string,
-    handleSave: PropTypes.func.isRequired,
-    handleView: PropTypes.func.isRequired,
-    isSaving: PropTypes.bool
-  };
+export default function SaveBar({
+  error,
+  handleSave,
+  handleView,
+  isSaving,
+  lastSaved,
+  pathForShowButton
+}) {
+  const isShowButtonRendered = handleView || pathForShowButton !== '/';
```

Comment:
Yeah, I think that's a great call –– also, then if people want a show button but don't want it to go anywhere, they can use '/' for the path to get it to show up. Bethany, would still love to hear your ideas!

Reply:
this is an interesting situation, where we're hiding the button but we're still trying to handle what happens if something calls the click handler anyway. I don't have a great example to offer, other than that we have typically preferred [conditional rendering](https://reactjs.org/docs/conditional-rendering.html) and so this problem doesn't come up. there's a lack of clear advice on the internet about when to use conditional rendering, but I would say our current most common practice is to prefer conditional rendering, unless you are optimizing for an expensive component that changes visibility a lot in which case using css to hide the component is cheaper.

I don't feel super strongly about preferring conditional rendering over css visibility per se, but if it would make it clearer that we could never hit the click handler when the component is hidden then it could be a good option in this case.

Reply:
if you do want to keep using css visibility then I would argue for NOT taking extra steps to handle the click in the case where the element is hidden -- if we think that hiding the css element doesn't really convince us that its not clickable then I would take that to mean that conditional rendering would be better.

Reply:
Good call––I was using visibility: hidden to be sure I wasn't causing unintentional styling changes. I've changed it to render an empty `<span \>` which is perhaps clunky, but (a) doesn't render a button if we don't need it, and (b) avoids the other buttons moving around.

Reply:
FYI for `ProgrammingExpressionEditor` the `/` path is being replaced in https://github.com/code-dot-org/code-dot-org/pull/43659/files, which I hope to merge today

Reply:
Excellent! I didn't want to change anything in that file because I knew you were updating it

Reply:
>Good call––I was using visibility: hidden to be sure I wasn't causing unintentional styling changes. I've changed it to render an empty <span \> which is perhaps clunky, but (a) doesn't render a button if we don't need it, and (b) avoids the other buttons moving around.

Ahh, yes, that is a very good reason to use css visibility which I had not considered. empty span sounds like a winner to me. thanks for all the thoughtful conversation here!